### PR TITLE
Fix #139 Add docker integration

### LINKED ISSUE #139: Docker integration
Similar to the support for ASP.NET
https://www.microsoft.com/en-us/download/details.aspx?id=46891

### PR DESCRIPTION
**Pull request by mousetraps**
- adds docker item template
- add integration point with Docker Tools

### CODE REVIEW

**Comment by billti** [comment]
Why are both the XML files showing up as being fully replaced? Is this just a line-ending thing? Makes it hard to see what has changed.

File: Nodejs/Product/Nodejs/Nodejs.csproj [hunk 1]
```diff
@@ -1597,6 +1604,7 @@
     </ItemGroup>
   </Target>
   <PropertyGroup>
-    <PostBuildEvent>copy /y $(OutDir)..\WebRole\Microsoft.NodejsTools.WebRole.dll $(OutDir)Debugger\RemoteDebug\</PostBuildEvent>
+    <PostBuildEvent>echo f | xcopy /y $(ProjectDir)Templates\Files\DockerfileTemplate\Dockerfile $(OutDir)Templates\Files\DockerfileTemplate\Dockerfile
```
Comment (billti) [inline Nodejs/Product/Nodejs/Nodejs.csproj]: What is the "echo f" for?

Comment (mousetraps) [inline Nodejs/Product/Nodejs/Nodejs.csproj]: xcopy will create new directories (unlike copy), but also asks whether it's a file or a directory (f or d), and there is no other way to suppress this.

**Comment by mousetraps** [comment]
Yes, it's a line ending thing (we really need to sweep through and fix all the files in one go to prevent this from continuing to happen)
you can use append ?w=1 to the url view it without the line endings
https://github.com/Microsoft/nodejstools/pull/280/files?w=1

**Comment by mousetraps** [comment]
Fixed the installer to add the loose dockerfile as well.

**Comment by mousetraps** [comment]
- adds docker item template
- add integration point with Docker Tools

**Pull request merged**